### ISSUE TITLE
Dynamic URIs with empty targets

### ISSUE THREAD

**Issue by rmatec** [Playtika/feign-reactive#239]
Hi,

We are seeing the following error when using an empty target and a URI in the interface.

```
Exception in thread "main" java.lang.UnsupportedOperationException: Empty targets don't have URLs
	at feign.Target$EmptyTarget.url(Target.java:166)
	at reactivefeign.methodhandler.PublisherClientMethodHandler.<init>(PublisherClientMethodHandler.java:75)
	at reactivefeign.methodhandler.ReactiveMethodHandlerFactory.create(ReactiveMethodHandlerFactory.java:34)
	at reactivefeign.ReactiveFeign.targetToHandlersByName(ReactiveFeign.java:126)
	at reactivefeign.ReactiveFeign.newInstance(ReactiveFeign.java:79)
	at reactivefeign.ReactiveFeignBuilder.target(ReactiveFeignBuilder.java:114)
```

To reproduce this, use the following code. It creates a standard feign client and a reactive one.
The former works as expected but the reactive one fails with the above exception.

```
public static void main(String[] args)
	{
		Test feign = Feign.builder()
		     .target(Target.EmptyTarget.create(Test.class));

                 try {
		    feign.send(URI.create("https://google.com"), "").block();
                 } catch (Exception e) {
                     e.printStackTrace();
                 }
		Test feignReactive = WebReactiveFeign
				.<Test>builder()
				.target(Target.EmptyTarget.create(Test.class));

		feignReactive.send(URI.create("https://google.com"), "").block();
	}

interface Test
{
	@RequestLine("PUT")
	Mono<Void> send(URI baseUrl, String data);

}
```

**Comment by jonathannaguin** [comment]
I used to be able to create a custom `Target` with no URL, like:
```
public class FeignDynamicTargetUrl<T> implements Target<T>
{
    private final Class<T> type;

    public FeignDynamicTargetUrl(Class<T> type) {
       this.type = type;
    }

    @Override
    public Class<T> type()
    {
        return type;
    }

    @Override
    public String name()
    {
        return "";
    }

    @Override
    public String url()
    {
        return "";
    }

    @Override
    public Request apply(RequestTemplate input)
    {
        return input.request();
    }
}
```

And then having the service contract defined as follows:
```
public interface GitHub
{
    @RequestLine("POST {baseUrl}")
    Mono<Issue> createIssue(@Param("baseUrl") URI baseUrl);
}
```

However, this has stopped working since version 2.0.0+, the URL is now being encoded by the library.

For example, in 1.0.46, the request line will look like:
```
POST https://example.com/event/?id=1111
```
In 2.0.0+, it looks like:
```
POST https://example.com/event/%3Fid=1111
```

**Comment by kptfh** [comment]
Hi guys. Working on it

**Comment by kptfh** [comment]
Released in 2.0,5, There is some gap in artifact publishing time in maven, 
Don't forget to put star. It will help to promote this project

**Comment by rmatec** [comment]
It works ok now.

Thanks.

PS: Starred
